### PR TITLE
Android deeplink sends "path + query" instead of just path

### PR DESCRIPTION
**Pull request by sarbagyastha**
## Description
Currently, android platform only sends URL path obtained from deeplink . This PR sets route as `path + query` instead of just path. 

The fix for iOS is in #23562  

## Related Issues

Fixes flutter/flutter#73637

## Tests

I have updated the test files.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Reviewer Checklist

- [ ] I have submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change.

### CODE REVIEW

**Comment by chunhtai** [comment]
hi @sarbagyastha thanks for putting up the PR, cc @johnpryan This looks something we have missed, do we need to update the doc regarding this change?

File: .gitignore [hunk 1]
```diff
@@ -4,10 +4,13 @@
 .*.sw?
 .DS_Store
 .ccls-cache
+.cipd/
```
Comment (chunhtai) [inline .gitignore:7]: can you revert change to this file?

**Review by chunhtai — CHANGES_REQUESTED** [review]
LGTM except the gitignore changes

File: .gitignore (hunk 1 above)
Comment (sarbagyastha) [reply .gitignore]: @chunhtai Done.

**Review by chunhtai — APPROVED** [review]

**Pull request merged**